src/main/java/edu/cornell/kfs/coa/document/CuAccountGlobalMaintainableImpl.java

```diff
@@ -45,6 +52,29 @@ public List<MaintenanceLock> generateMaintenanceLocks() {
         return Stream.concat(accountLocks.stream(), trickleDownLocks)
                 .collect(Collectors.toUnmodifiableList());
     }
+    
+    @Override
+    public void doRouteStatusChange(DocumentHeader documentHeader) {
+        super.doRouteStatusChange(documentHeader);
+        if (MaintenanceUtils.shouldClearCacheOnStatusChange(documentHeader)) {
+            clearBlockingCacheForCurrentDocument();
+        }
+    }
+
+    public void clearBlockingCacheForCurrentDocument() {
+        Cache cache = MaintenanceUtils.getBlockingCache();
+        String cacheKey = MaintenanceUtils.buildLockingDocumentCacheKey(getDocumentNumber());
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("clearBlockingCacheForCurrentDocument, clear cache id: " + cacheKey);
+        }
+        cache.evictIfPresent(cacheKey);
+    }
+    
+    @Override
+    public void prepareForSave() {
+        clearBlockingCacheForCurrentDocument();
```

Comment:
Can you please confirm if you're still seeing the expected performance gains locally or in Pre-KEW? If you're putting a cache-clearing operation in the `prepareForSave()` method, I'm wondering if that would cause the cache to be cleared too frequently and potentially nullify the intended performance improvements.

Reply:
my pre-kew testing has good performance results

Reply:
Thanks for confirming. I'm okay with moving the changes forward to DEV for more in-depth validation.